packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/util/ClientModelUtil.java

```diff
@@ -879,6 +879,31 @@ public static boolean readOnlyNotInCtor(ClientModel model, ClientModelProperty p
                 && !(property.isRequired() && settings.isRequiredFieldsAsConstructorArgs()));
     }
 
+    // If stream-style serialization is being generated, some additional setters may need to be added to
+    // support read-only properties that aren't included in the constructor.
+    // Jackson handles this by reflectively setting the value in the parent model, but stream-style
+    // serialization doesn't perform reflective cracking like Jackson Databind does, so it needs a way
+    // to access the readonly property (aka one without a public setter method).
+    //
+    // The package-private setter is added when the property isn't included in the constructor and is
+    // defined by this model, except for JSON merge patch models as those use the access helper pattern
+    // to enable subtypes to set the property.
+    public static boolean needsPackagePrivateSetter(ClientModel model, ClientModelProperty property,
```

Comment:
similarly, a better name?

Reply:
It applies to all field properties. It's extracted from https://github.com/microsoft/typespec/pull/5472/files#r1901449091

Reply:
Enhanced javadoc.